pkg/resources/nodeagent/nodeagent.go

```diff
@@ -304,6 +304,10 @@ func (r *Reconciler) Reconcile() (*reconcile.Result, error) {
 			if err != nil {
 				return nil, err
 			}
+
+			// Overwrite Kubernetes endpoint with a ClusterDomain templated value.
+			NodeAgentFluentbitDefaults.FluentbitSpec.FilterKubernetes.KubeURL = fmt.Sprintf("https://kubernetes.default.svc.%s:443", r.Logging.Spec.ClusterDomain)
```

Comment:
I think this format string will have the same problem with the trailing dot if `ClusterDomain` is set to empty as I noted earlier with the chart template.